### DISCUSSION TITLE
Setting up an IP Pool. Do I include the main IP address?

### DISCUSSION THREAD

**Discussion by Dave777** [postalserver/postal#2725] [Help with using Postal]
Hi,
I hope someone can help clarify the DNS for me :)
I am using OVH and they give me the first IP address with the vps of course.
Then I have 3 extra IP addresses.

When creating an IP Pool, I want to rotate equally through all 4 IP addresses. For now I will only create one pool.

So I guess I put the main IP address, which has sub-domain of postal.mydomain.com into the pool

So:
IP4  xxx.xxx.xxx.1  HOSTNAME:  postal.mydomain.com

Then for the additional IP addresses, should I have separate sub-domains?
EG
IP4  xxx.xxx.xxx.2  HOSTNAME:  postal2.mydomain.com
IP4  xxx.xxx.xxx.3  HOSTNAME:  postal3.mydomain.com
IP4  xxx.xxx.xxx.4  HOSTNAME:  postal4.mydomain.com

OR should I have the additional IP addresses all with same sub-domain ?

EG
IP4  xxx.xxx.xxx.1  HOSTNAME:  postal.mydomain.com
IP4  xxx.xxx.xxx.2  HOSTNAME:  postal.mydomain.com
IP4  xxx.xxx.xxx.3  HOSTNAME:  postal.mydomain.com
IP4  xxx.xxx.xxx.4  HOSTNAME:  postal.mydomain.com

Which is the proper setup?

And now do I have to change ALL the SPF records?

EG
TXT  mydomain.com   v=spf1 +a +mx +ip4:xxx.xxx.xxx.1 +ip4:xxx.xxx.xxx.2 +ip4:xxx.xxx.xxx.3 +ip4:xxx.xxx.xxx.4 include:spf.postal.mydomain.com ~all

TXT  rp.postal   v=spf1 +a +mx +ip4:xxx.xxx.xxx.1 +ip4:xxx.xxx.xxx.2 +ip4:xxx.xxx.xxx.3 +ip4:xxx.xxx.xxx.4 include:spf.postal.mydomain.com ~all

TXT  spf.postal   v=spf1 +a +mx +ip4:xxx.xxx.xxx.1 +ip4:xxx.xxx.xxx.2 +ip4:xxx.xxx.xxx.3 +ip4:xxx.xxx.xxx.4 include:spf.postal.mydomain.com ~all

Are the above correct - The "+" symbols - are they correct?

It's a bit confusing !

**Comment by willpower232** [comment]
based on https://serverfault.com/a/499786 I believe that the pluses are supposed to convey some extra meaning however I have never seen them before so I don't think they matter as much as choosing between `~all` and `-all`.

The IP addresses referred to by SPF records have to be the ones sending the email so if you're intending to send email from them, they need to feature in the SPF record for the domain you're sending from.

You could also move the extra IP addresses to their own SPF record and include it so heres a more complete example

txt record | value 
---|----
spf.postal.my-company.com | `v=spf1 ip4:xxx.xxx.xxx.1 -all`
ip-pool.postal.my-company.com |  `v=spf1 ip4:xxx.xxx.xxx.2 ip4:xxx.xxx.xxx.3 ip4:xxx.xxx.xxx.4 -all`

Now you could do

txt record | value 
---|----
my-company.com | `v=spf1 include:spf.postal.mydomain.com -all`
my-client.com | `v=spf1 include:ip-pool.postal.my-company.com -all`

I don't think you would need to include all the IP addresses in the rp.postal SPF record but I could be wrong so definitely test with https://mail-tester.com before you get too deep.

**Comment by Dave777** [comment]
Thanks for your response.
I am still struggling a bit :(

I am using Cloudflare for the DNS.

My records look like this : 
![chrome_U51hdDqj9Y](https://github.com/postalserver/postal/assets/1362368/a16329ed-fc10-4572-b867-903f7b6f625a)

So should I be adding another A record for each additional IP address ?
EG

A   postal  123.98.124.02
A   postal  123.98.124.03
A   postal  123.98.124.04

And should these be proxied or DNS only ?

And I am not sure why I am getting those Red Triangle warnings , 
The DNS-only ose say hat the IP address is exposed.
The Proxied ones say the hostname is not covered by a certificate.

Thanks for helping.

**Reply by willpower232** [reply]
I wouldn't bother using Cloudflare proxying on Postal, it would only add confusion. DNS only should be fine.

The three IP addresses you want to send from need to be in the SPF record of the domain you want to use with them so however that happens, it doesn't need to touch your main Postal DNS records in this screenshot at all.

**Reply by Dave777** [reply]
Thanks Will,
However in my OVH settings I could not set the Reverse DNS to the "postal" sub directory for the additional IP addresses  
the reverse DNS is stuck on "server".

![msedge_mMtJsZsxF1](https://github.com/postalserver/postal/assets/1362368/b4331ef7-8006-4445-88ea-7c76c2d4b56c)

and when I asked support they said this:
################################################
Thanks for reaching us David.

Actually you're declarying your A record for your server 192.99.167.139.

As you can see, there are no additional IPs declared rather than your IP [192.99.167.139](http://192.99.167.139/):

dig [postal.mailserve2.com](http://postal.mailserve2.com/)

; <<>> DiG 9.10.3-P4-Debian <<>> [postal.mailserve2.com](http://postal.mailserve2.com/)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 32530
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

#############################################

So, it looks to me like I need to add the additional A records, No?

**Reply by willpower232** [reply]
Yeah the IP addresses need to point to the server (and have reverse DNS) otherwise you will probably have some mail sending issues.
